wgc/wgc_wgni.py

```diff
@@ -341,6 +348,10 @@ def __oauth_challenge_calculate(self, challenge_data):
                 return pow_number
 
             pow_number = pow_number + 1
+            
+            #prevent application become unresponsive
+            if pow_number % 100 == 0:
+                await asyncio.sleep(0)
```

Comment:
the best practice would be to call synchronous `__oauth_changllenge_calculate` in separate thread/process.
eg. by using `asyncio.run_in_executor`. See https://docs.python.org/3.7/library/asyncio-dev.html#running-blocking-code


(well, in fact the same is applicable for local sizes)

Reply:
@Mixaill